src/transformers/modeling_utils.py

```diff
@@ -2199,10 +2326,6 @@ def _find_mismatched_keys(
                         del state_dict[checkpoint_key]
             return mismatched_keys
 
-        if low_cpu_mem_usage:
-            model_state_dict = None  # free references to model's params to allow memory freeing
-            _move_model_to_meta(model, loaded_keys, start_prefix)
```

Comment:
are you sure this removal doesn't end up keeping a reference count up and not freeing the memory? It took me quite some time to hunt this one down.

I hope the test I wrote would have caught the problem, but it was tricky to write as results vary from machine to machine, so want to make sure we validate this removal. Thank you for bearing with me.

Reply:
At this stage, the model is already on the meta device since it was created under the `no_init_weights` context manager. The `model_state_dict` is thus a dictionary param name to empty tensors on the meta device, which doesn't take any CPU memory.

As more tests, I have been able to load and use T0pp with the code snippet above, on my machine which does not have enough RAM for the whole model (32GB while the model needs 40), does that ease your mind? What more tests can I add to ease it further?